### PR TITLE
Update Hack The Box domain

### PR DESCRIPTION
**Pull request by sn00py1310**
The Hack The Box domain changed from `hackthebox.eu` -> `hackthebox.com`
Hack the Box logins are possible from the `app` and `www` subdomain.

Hack The Box and Hack The Box _**Forums**_ have for some reason a different login system. 🤷

### Forums prove
Forums 2FA

### CODE REVIEW

File: entries/a/app.hackthebox.com.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "Hack The Box": {
+    "domain": "app.hackthebox.com",
+    "additional-domains": [
+      "app.hackthebox.eu",
+      "www.hackthebox.com",
+      "www.hackthebox.eu"
```
Comment (Carlgo11) [inline entries/a/app.hackthebox.com.json]: subdomains shouldn't be specified for domains.
```suggestion
      "hackthebox.eu"
```

File: entries/a/app.hackthebox.com.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
+{
+  "Hack The Box": {
+    "domain": "app.hackthebox.com",
```
Comment (Carlgo11) [inline entries/a/app.hackthebox.com.json]:
```suggestion
    "domain": "hackthebox.com",
```

File: entries/f/forum.hackthebox.com.json [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    "tfa": [
+      "totp",
+      "u2f"
+    ],
+    "keywords": [
+      "education",
+      "social"
+    ]
+  }
+}
```
Comment (Carlgo11) [inline entries/f/forum.hackthebox.com.json]: We don't list forums.
```suggestion
```

**Review by Carlgo11 — CHANGES_REQUESTED** [review]

**Comment by sn00py1310** [comment]
@Carlgo11 is there a reason why forums aren't allowed. I didn't find anything in the exclusions.

**Comment by Carlgo11** [comment]
It's been a rule for as long as I've worked on the project.
Some of the problems with listing forums are:
* There's no good way of ranking the popularity as nearly all use subdomains and thus piggyback on the domain's Alexa rank.
* The impact of having your forum account stolen is for most people negligible.
* Most violate the self-hosting exception.

You're right that we should list that rule somewhere. I'm not sure why that hasn't been done already :thinking:

**Review by Carlgo11 — APPROVED** [review]

**Review by ApCoder123 — APPROVED** [review]

**Pull request merged**